packages/contracts-periphery/test/contracts/standards/L2StandardERC721.spec.ts

```diff
@@ -0,0 +1,118 @@
+/* External Imports */
+import { ethers } from 'hardhat'
+import { Signer, Contract } from 'ethers'
+import { smock, FakeContract } from '@defi-wonderland/smock'
+
+/* Internal Imports */
+import { expect } from '../../setup'
+
+const TOKEN_ID = 10
+const DUMMY_L1ERC721_ADDRESS: string =
+  '0x2234223412342234223422342234223422342234'
+
+describe('L2StandardERC721', () => {
+  let l2BridgeImpersonator: Signer
+  let alice: Signer
+  let Fake__L2ERC721Bridge: FakeContract
+  let L2StandardERC721: Contract
+  let l2BridgeImpersonatorAddress: string
+  let aliceAddress: string
+  let baseUri: string
+  let chainId: number
+
+  before(async () => {
+    ;[l2BridgeImpersonator, alice] = await ethers.getSigners()
+    l2BridgeImpersonatorAddress = await l2BridgeImpersonator.getAddress()
+    aliceAddress = await alice.getAddress()
+
+    chainId = await alice.getChainId()
+    baseUri = ''.concat(
+      'ethereum:',
+      DUMMY_L1ERC721_ADDRESS,
+      '@',
+      chainId.toString(),
+      '/tokenURI?uint256='
+    )
+
+    L2StandardERC721 = await (
+      await ethers.getContractFactory('L2StandardERC721')
+    ).deploy(
+      l2BridgeImpersonatorAddress,
+      DUMMY_L1ERC721_ADDRESS,
+      'L2ERC721',
+      'ERC',
+      { gasLimit: 4_000_000 } // Necessary to avoid an out-of-gas error
+    )
+
+    // Get a new fake L2 bridge
+    Fake__L2ERC721Bridge = await smock.fake<Contract>(
+      'L2ERC721Bridge',
+      // This allows us to use an ethers override {from: Fake__L2ERC721Bridge.address} to mock calls
+      { address: await l2BridgeImpersonator.getAddress() }
+    )
+
+    // mint an nft to alice
+    await L2StandardERC721.connect(l2BridgeImpersonator).mint(
+      aliceAddress,
+      TOKEN_ID,
+      {
+        from: Fake__L2ERC721Bridge.address,
+      }
+    )
+  })
+
+  describe('constructor', () => {
+    it('should be able to create a standard ERC721 contract with the correct parameters', async () => {
+      expect(await L2StandardERC721.l2Bridge()).to.equal(
+        l2BridgeImpersonatorAddress
+      )
+      expect(await L2StandardERC721.l1Token()).to.equal(DUMMY_L1ERC721_ADDRESS)
+      expect(await L2StandardERC721.name()).to.equal('L2ERC721')
+      expect(await L2StandardERC721.symbol()).to.equal('ERC')
+      expect(await L2StandardERC721.baseTokenURI()).to.equal(baseUri)
+
+      // alice has been minted an nft
+      expect(await L2StandardERC721.ownerOf(TOKEN_ID)).to.equal(aliceAddress)
+    })
+  })
+
+  describe('mint and burn', () => {
```

Comment:
Could you add tests for a pranked call that mints/burns and the expected events are emitted?

Reply:
Added these tests to L2ERC721Bridge.spec.ts. I figured that they didn't need to be repeated here, but I can add them if you'd like :)